### PR TITLE
layer_name property getter and setter fix

### PR DESCRIPTION
**Pull request by gkorompi**
layer_name bug fix. Please compare previous and new code.

### CODE REVIEW

File: tests/legacy/system/test_edb_modeler.py [hunk 1]
```diff
@@ -323,3 +323,58 @@ def test_156_check_path_length(self):
...
+        from pyedb.dotnet.edb import Edb
+        edbapp = Edb()
+        edbapp["$H"] = "0.65mil"
+        assert edbapp["$H"].value_string == "0.65mil"
+        edbapp.stackup.add_layer("bot_gnd", thickness="0.65mil")
+        edbapp.stackup.add_layer("d1", layer_type="dielectric", thickness="$S_D", material="FR4_epoxy")
+        edbapp.stackup.add_layer("trace2", thickness="$H")
+        edbapp.stackup.add_layer("d2", layer_type="dielectric", thickness="$T-$S_D", material="FR4_epoxy")
+        edbapp.stackup.add_layer("mid_gnd", thickness="0.65mil")
+        edbapp.stackup.add_layer("d3", layer_type="dielectric", thickness="13mil", material="FR4_epoxy")
```
Comment (svandenb-dev) [inline tests/legacy/system/test_edb_modeler.py:373]: You are missing the assert in your test

File: src/pyedb/dotnet/edb_core/edb_data/primitives_data.py [hunk 2]
```diff
@@ -1005,6 +1005,35 @@ def clone(self):
...
+            if lay in self._pedb.stackup.layers:
+                dupli_poly = self._app.edb_api.cell.primitive.polygon.create(
+                    self._app.active_layout, lay, self.net, self.polygon_data.edb_api
+                )
+                if dupli_poly:
+                    for void in self.voids:
+                        dupli_void = self._app.edb_api.cell.primitive.polygon.create(
+                            self._app.active_layout, lay, self.net, void.polygon_data.edb_api
+                        )
+                        dupli_poly.prim_obj.AddVoid(dupli_void.prim_obj)
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/primitives_data.py:1032]:
```suggestion
        for layer in layers:
            if layer in self._pedb.stackup.layers:
                dupli_poly = self._app.edb_api.cell.primitive.polygon.create(
                    self._app.active_layout, layer, self.net, self.polygon_data.edb_api
                )
                if dupli_poly:
                    for void in self.voids:
                        dupli_void = self._app.edb_api.cell.primitive.polygon.create(
                            self._app.active_layout, layer, self.net, void.polygon_data.edb_api
                        )
                        dupli_poly.prim_obj.AddVoid(dupli_void.prim_obj)
```

File: src/pyedb/dotnet/edb_core/edb_data/primitives_data.py [hunk 3]
```diff
@@ -1005,6 +1005,35 @@ def clone(self):
             return cloned_poly
         return False
 
+    @pyedb_function_handler()
+    def duplicate_across_layers(self, layers):
+        """Duplicate across layer a primitive object.
+
+        Parameters:
+
+        layer: list
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/primitives_data.py:1014]:
```suggestion
        layers: list
```

File: src/pyedb/dotnet/edb_core/edb_data/primitives_data.py [hunk 4]
```diff
@@ -135,15 +135,15 @@ def layer_name(self):
         except AttributeError:  # pragma: no cover
             return None
 
-    @layer_name.setter
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/primitives_data.py:138]: You just removed the fact that property decorates with `.setter`. You should put back the removed `@layer_name.setter`

File: tests/legacy/system/test_edb_modeler.py [hunk 5]
```diff
@@ -327,3 +327,58 @@ def test_156_check_path_length(self):
...
+        assert c1
+        c2 = edbapp.modeler.create_circle(layer_name="top_gnd", x="-Via_S/2", y="0mil", radius="$Antipad_R", net_name="c2")
+        assert c2
+        r = [r2, c1, c2]
+        assert r1.subtract(r)
+        lay_list = ["bot_gnd", "mid_gnd"]
+        edbapp.modeler.primitives_by_layer
+        assert edbapp.modeler.primitives[0].duplicate_across_layers(lay_list)
+        if edbapp.modeler.primitives_by_layer["mid_gnd"]: print("True")
+        if edbapp.modeler.primitives_by_layer["bot_gnd"]: print("True")
```
Comment (SMoraisAnsys) [inline tests/legacy/system/test_edb_modeler.py:363]: Please use assertions here instead of print

File: tests/legacy/system/test_edb_modeler.py [hunk 6]
```diff
@@ -327,3 +327,58 @@ def test_156_check_path_length(self):
...
+        edbapp.stackup.add_layer("trace2", thickness="$H")
+        edbapp.stackup.add_layer("d2", layer_type="dielectric", thickness="$T-$S_D", material="FR4_epoxy")
+        edbapp.stackup.add_layer("mid_gnd", thickness="0.65mil")
+        edbapp.stackup.add_layer("d3", layer_type="dielectric", thickness="13mil", material="FR4_epoxy")
+        edbapp.stackup.add_layer("top_gnd", thickness="0.65mil")
+        edbapp.stackup.add_layer("d4", layer_type="dielectric", thickness="13mil", material="FR4_epoxy")
+        edbapp.stackup.add_layer("trace1", thickness="$H")
+        r1 = edbapp.modeler.create_rectangle(center_point=("0,0"), width="200mil", height="200mil", layer_name="top_gnd", representation_type="CenterWidthHeight", net_name="r1")
+        r1.layer_name = "mid_gnd"
+        r1.layer_name
```
Comment (SMoraisAnsys) [inline tests/legacy/system/test_edb_modeler.py:383]: As @svandenb-dev said, you are missing assertions here.

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Thanks for the PR, I would advise to rework the test in the PR to correct the lack of assert(s) and make them simpler. I have the feeling that the created data does not need to be so complex. Moreover, since you are adding something to the test class, it might be wise to leverage what is already performed in other tests. 

Note: I'm clearly in favor of having tests that can run in standalone (like you did) but it seems that opening and closing EDB a lot of times can lead to unexpected failures.

**Review by SMoraisAnsys — COMMENTED** [review]
LGTM

Please, next time, commit the suggestion as proposed when possible. It eases the fact of reviewing again.

**Review by svandenb-dev — DISMISSED** [review]
LGTM

**Review by SMoraisAnsys — APPROVED** [review]
LGTM

**Pull request closed without merging**